qiskit/pulse/frame.py

```diff
@@ -0,0 +1,77 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Implements a Frame."""
+
+from typing import Optional, Tuple
+
+from qiskit.circuit import Parameter
+from qiskit.pulse.utils import validate_index
+from qiskit.pulse.exceptions import PulseError
+
+
+class Frame:
```

Comment:
Do `Frame` not inherit from the root channel or instruction type? They are their own entity in the IR?

Reply:
I don't think a frame can be seen as an instruction. Seeing it as a channel seems to stretch the definition of a channel.